src/protocol/sort/apply.rs

```diff
@@ -11,77 +12,121 @@ use permutation::Permutation;
 
 #[embed_doc_image("apply", "images/sort/apply.png")]
 #[embed_doc_image("apply_inv", "images/sort/apply_inv.png")]
+#[embed_doc_image("apply", "images/sort/apply.png")]
+#[embed_doc_image("apply_inv", "images/sort/apply_inv.png")]
+
 /// Permutation reorders (1, 2, . . . , m) into (σ(1), σ(2), . . . , σ(m)).
 /// For example, if σ(1) = 2, σ(2) = 3, σ(3) = 1, and σ(4) = 0, an input (A, B, C, D) is reordered into (C, D, B, A) by σ.
 /// ![Apply steps][apply]
-pub fn apply<T: Copy + Default, S>(mut permutation: Permutation, values: &mut S)
-where
-    S: AsMut<[T]>,
-{
-    permutation.apply_slice_in_place(values);
+pub fn apply<T: Copy + Default>(permutation: &[usize], values: &mut [T]) {
```

Comment:
I think there is a way to require `T: Clone` only. That is probably worth doing because of #120

Reply:
I'll give it a try.